docs/tutorials/nextjs.mdx

```diff
@@ -0,0 +1,186 @@
+
+### How to Read This Tutorial
+
+This page assumes that you already have an exisiting NextJS application based on the App Router architecture.
+
+## The App Router Architecture and Redux
+
+The primary new feature of the NextJS App Router is the addition of support for React Server Components (RSCs). RSCs are a special type of React component that only renders on the server, as opposed to "client" components that render on **both** the client and the server. RSCs can be defined as `async` functions and return promises during rendering as they make async requests for data to render.
+
+RSCs abilitiy to block for data means that with the App Router you no longer have `getServerSideProps` to fetch data for rendering. Any component in the tree can make asychronous requests for data. While this is very convenient it also means thats if you define global variables (like the Redux store) they will be shared across requests. This is a problem because the Redux store could be contaminated with data from other requests.
+
+### Creating a Redux Store per Request
+
+Following along with the (Quick-Start guide)[./quick-start.mdx] we need to make some changes to the `app/store.js` file. The first change is to move from defining store as a global to defining a `createStore` function that returns a new store for each request.
+
+```ts title="src/app/store.ts"
+import { configureStore } from '@reduxjs/toolkit'
+
+export const createStore = () =>
+  configureStore({
+    reducer: {},
+  })
+
+// Infer the type of createStore
+export type StoreType = ReturnType<typeof createStore>
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<StoreType['getState']>
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
+export type AppDispatch = StoreType['dispatch']
+```
+
+To use this new `createStore` function we need to create a new "client" component that will create the store and share it using the React-Redux `Provider` component.
+
+```ts title="src/app/StoreProvider.tsx"
+// file: app/store.ts noEmit
+import { configureStore } from '@reduxjs/toolkit'
+
+// highlight-start
+export const createStore = () =>
+  configureStore({
+    reducer: {},
+  })
+// highlight-end
+
+// Infer the type of createStore
+// highlight-start
+export type StoreType = ReturnType<typeof createStore>
+// highlight-end
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<StoreType['getState']>
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
+export type AppDispatch = StoreType['dispatch']
+
+// file: app/StoreProvider.tsx
+;('use client')
```

Comment:
can we get prettier to work without changing the pragma into this form? It's probably not reasily recognizable like this.